From 2f2134cc43220ca1229e9f4a63a48d5c458d23fc Mon Sep 17 00:00:00 2001
From: Desnes Nunes <desnesn@linux.ibm.com>
Date: Mon, 9 May 2022 17:52:36 -0300
Subject: [PATCH 20/23] beaker: dn: ubuntu: adapt test_netboot.py to ubuntu's semicolon kernel options

Ubuntu's support came with the necessity of parsing ; into ' ' on grub2
options. This breaks a few rpmbuild tests, which are being fixed here by
adding the ' ' to the grub2 tests.

Fixes: 5f631ddd32c1 ("beaker: dn: ubuntu: Quote kernel arguments in grub.conf")
Signed-off-by: Desnes Nunes <desnesn@linux.ibm.com>
---
 LabController/src/bkr/labcontroller/test_netboot.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/LabController/src/bkr/labcontroller/test_netboot.py b/LabController/src/bkr/labcontroller/test_netboot.py
index b86035574..d6bc36928 100644
--- a/LabController/src/bkr/labcontroller/test_netboot.py
+++ b/LabController/src/bkr/labcontroller/test_netboot.py
@@ -533,7 +533,7 @@ def test_configure_then_clear(self):
                               os.path.join(self.tftp_root, 'grub.cfg-7F0000FF')]
         for path in grub2_configs_path:
             self.assertEquals(open(path).read(), """\
-linux  /images/fqdn.example.invalid/kernel console=ttyS0,115200 ks=http://lol/ netboot_method=grub2
+linux  /images/fqdn.example.invalid/kernel 'console=ttyS0,115200' 'ks=http://lol/' netboot_method=grub2
 initrd /images/fqdn.example.invalid/initrd
 
 boot
@@ -561,7 +561,7 @@ def test_configure_then_clear(self):
 
         for path in grub2_configs_path:
             self.assertEquals(open(path).read(), """\
-linux  /images/fqdn.example.invalid/kernel console=ttyS0,115200 ks=http://lol/ netboot_method=grub2
+linux  /images/fqdn.example.invalid/kernel 'console=ttyS0,115200' 'ks=http://lol/' netboot_method=grub2
 initrd /images/fqdn.example.invalid/initrd
 
 boot
@@ -588,7 +588,7 @@ def test_configure_then_clear(self):
         grub_config_path = os.path.join(self.tftp_root, 'aarch64', 'grub.cfg-7F0000FF')
         grub_default_path = os.path.join(self.tftp_root, 'aarch64', 'grub.cfg')
         self.assertEquals(open(grub_config_path).read(), """\
-linux  /images/fqdn.example.invalid/kernel console=ttyS0,115200 ks=http://lol/ netboot_method=grub2
+linux  /images/fqdn.example.invalid/kernel 'console=ttyS0,115200' 'ks=http://lol/' netboot_method=grub2
 initrd /images/fqdn.example.invalid/initrd
 
 boot
@@ -604,7 +604,7 @@ def test_alternate_devicetree(self):
                                   'devicetree=custom.dtb ks=http://lol/', self.tftp_root)
         grub_config_path = os.path.join(self.tftp_root, 'aarch64', 'grub.cfg-7F0000FF')
         self.assertEquals(open(grub_config_path).read(), """\
-linux  /images/fqdn.example.invalid/kernel ks=http://lol/ netboot_method=grub2
+linux  /images/fqdn.example.invalid/kernel 'ks=http://lol/' netboot_method=grub2
 initrd /images/fqdn.example.invalid/initrd
 devicetree custom.dtb
 boot
-- 
2.35.3

